etl/exporter/record_exporter_server.go

```diff
@@ -5,13 +5,48 @@
 
 package exporter
 
+import (
+	"context"
+	"fmt"
+
+	"github.com/insolar/assured-ledger/ledger-core/v2/log"
+
+	"github.com/insolar/block-explorer/configuration"
+	"github.com/insolar/block-explorer/etl/interfaces"
+	"github.com/insolar/block-explorer/instrumentation/belogger"
+)
+
 type RecordServer struct {
+	storage interfaces.StorageExporterFetcher
+	logger  log.Logger
+	config  configuration.Exporter
 }
 
-func NewRecordServer() *RecordServer {
-	return &RecordServer{}
+func NewRecordServer(ctx context.Context, storage interfaces.StorageExporterFetcher, config configuration.Exporter) *RecordServer {
+	logger := belogger.FromContext(ctx)
+	return &RecordServer{storage: storage, logger: logger, config: config}
 }
 
-func (s *RecordServer) GetRecords(*GetRecordsRequest, RecordExporter_GetRecordsServer) error {
+func (s *RecordServer) GetRecords(request *GetRecordsRequest, stream RecordExporter_GetRecordsServer) error {
+	recs, err := s.storage.GetRecordsByPrototype(request.Prototypes, request.PulseNumber, request.Count, request.RecordNumber)
+	if err != nil {
+		return fmt.Errorf("error on requesting records")
```

Comment:
what about returning the error do log it here?

Reply:
yes